web/src/filter.ts

```diff
@@ -1151,6 +1151,22 @@ export class Filter {
             return true;
         }
 
+        if (_.isEqual(term_types, ["not-is-dm"])) {
+            return true;
+        }
+
+        if (_.isEqual(term_types, ["channel", "not-is-dm"])) {
+            return true;
+        }
+
+        if (_.isEqual(term_types, ["topic", "not-is-dm"])) {
+            return true;
+        }
+
+        if (_.isEqual(term_types, ["channel", "topic", "not-is-dm"])) {
+            return true;
+        }
```

Comment:
The function is called `single_term_type_returns_all_messages_of_conversation`; is that the right only function in this file to change?